frontend/lib/modules/user_auth/current_user_state.dart

```diff
@@ -73,27 +75,54 @@ class CurrentUserState extends ChangeNotifier {
 
   void checkAndLogin() {
     getLocalstorage();
-    var user = _localstorage?.getItem('currentUser');
+    Map<String, dynamic>? _localStorageUser = _localstorage?.getItem('currentUser');
+    UserClass? user = _localStorageUser != null? UserClass.fromJson(_localStorageUser):null;
     if (user != null) {
       _status = "loading";
-      _socketService.emit('getUserSession', { 'user_id': user['id'], 'session_id': user['session_id'] });
+      _socketService.emit('getUserSession', {  'user_id': user.id, 'session_id': user.session_id });
+      _currentUser = user;
+      _isLoggedIn = true;
     }
   }
 
   void logout() {
     if (_currentUser != null) {
       _status = "loading";
-      _socketService.emit('logout', { 'user_id': _currentUser.id, 'session_id': _currentUser.session_id });
+      _socketService.emit('logout', { 'user_id': _currentUser!.id, 'session_id': _currentUser!.session_id });
     }
   }
 
+  Future<List<dynamic>> getUserLocation() async {
+    List<dynamic> _lngLat = [];
+    LocalStorage _localStorage = _localstorageService.localstorage;
+    List<dynamic>? _lngLatLocalStored = _localStorage.getItem('lngLat');
```

Comment:
We already have this stored in the user data so we can pull it from there (and that is stored in localstorage too). So we can just do `currentUser.lngLat`

Reply:
Yes, but for unregistered users, it would still load very slow every time the get to the lend library page if we don't store lngLat locally separately